vizro-core/tests/tests_utils/e2e_asserts.py

```diff
@@ -0,0 +1,56 @@
+import shutil
+from pathlib import Path
+
+import cv2
+import imutils
+from hamcrest import assert_that, equal_to
+
+
+def _compare_images(expected_image, result_image):
+    """Comparison process."""
+    difference = cv2.subtract(expected_image, result_image)
+    blue, green, red = cv2.split(difference)
+    assert_that(cv2.countNonZero(blue), equal_to(0), reason="Blue channel is different")
+    assert_that(cv2.countNonZero(green), equal_to(0), reason="Green channel is different")
+    assert_that(cv2.countNonZero(red), equal_to(0), reason="Red channel is different")
+
+
+def _create_image_difference(expected_image, result_image):
+    """Creates new image with diff of images comparison."""
+    diff = expected_image.copy()
+    cv2.absdiff(expected_image, result_image, diff)
+    gray = cv2.cvtColor(diff, cv2.COLOR_BGR2GRAY)
+    for i in range(0, 3):
+        dilated = cv2.dilate(gray.copy(), None, iterations=i + 1)
+    (t_var, thresh) = cv2.threshold(dilated, 3, 255, cv2.THRESH_BINARY)
+    cnts = cv2.findContours(thresh, cv2.RETR_LIST, cv2.CHAIN_APPROX_SIMPLE)
+    cnts = imutils.grab_contours(cnts)
+    for contour in cnts:
+        (x, y, width, height) = cv2.boundingRect(contour)
+        cv2.rectangle(result_image, (x, y), (x + width, y + height), (0, 255, 0), 2)
+    return result_image
+
+
+def make_screenshot_and_paths(browserdriver, request_node_name):
+    result_image_path = f"{request_node_name}_branch.png"
+    expected_image_path = f"tests/e2e/screenshots/{request_node_name.replace('test', 'main')}.png"
+    browserdriver.save_screenshot(result_image_path)
+    return result_image_path, expected_image_path
+
+
+def assert_image_equal(result_image_path, expected_image_path):
+    """Comparison logic and diff files creation."""
+    expected_image = cv2.imread(expected_image_path)
+    expected_image_name = Path(expected_image_path).name
+    result_image = cv2.imread(result_image_path)
+    try:
+        _compare_images(expected_image, result_image)
+        Path(result_image_path).unlink()
+    except AssertionError as exc:
+        shutil.copy(result_image_path, expected_image_name)
+        diff = _create_image_difference(expected_image=expected_image, result_image=result_image)
+        cv2.imwrite(f"{result_image_path}_difference_from_main.png", diff)
+        raise AssertionError("pictures are not the same") from exc
+    except cv2.error as exc:
```

Comment:
it could be `cv2.error` in `try` in this line:
https://github.com/mckinsey/vizro/blob/bbe76f4c96f2041fc32f84c5acc85b2547a65bea/vizro-core/tests/tests_utils/e2e_asserts.py#L11

Reply:
but where is the `try` statement, sorry if I am being think

Reply:
here it is:

https://github.com/mckinsey/vizro/blob/bbe76f4c96f2041fc32f84c5acc85b2547a65bea/vizro-core/tests/tests_utils/e2e_asserts.py#L46

Reply:
I see, I had never used a double except myself actually, learned something, thanks :)